src/redisjson.rs

```diff
@@ -52,16 +52,35 @@ impl RedisJSON {
     pub fn to_string(&self, path: &str) -> Result<String, Error> {
         eprintln!("Serializing back to JSON");
 
+        let s = match self.get_doc(path)? {
+            Some(doc) => serde_json::to_string(&doc)?,
+            None => String::new()
+        };
+
+        Ok(s)
+    }
+
+    pub fn str_len(&self, path: &str) -> Result<usize, Error> {
+        let s = match self.get_doc(path)? {
+            Some(doc) => {
+                if doc.is_string() {
+                    serde_json::to_string(&doc)?.len() - 2 // removes ""
```

Comment:
You don't need to use `serde_json::to_string` here: That is meant to serialize any value *to* JSON.
You already have JSON here: If the underlying value is a `String`, you can get it with `doc.as_str()`.
This way you also don't get the extra quotes, so no need to remove them.

This way, the whole `if/else` can be replaced by:
```
doc.as_str().map_or(0, |d| d.len());
```

But see the next comment.